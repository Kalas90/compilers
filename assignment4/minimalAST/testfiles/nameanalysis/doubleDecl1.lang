int main(int d, int duplParam, int duplParam) {
	int d;
	int a = 1;
	duplParam = 1;
	int a = 2;
	int b = 1;
	int a = 1;
	a = 20;
	int b;
	d = 10;
	if(a == 10) {
		int a = 10;
		int c = 1;
	}	
	c = 2;
}
int main() {
}

