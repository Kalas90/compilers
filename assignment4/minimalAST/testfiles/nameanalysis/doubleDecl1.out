Error at line 1: symbol 'main' is already declared!
