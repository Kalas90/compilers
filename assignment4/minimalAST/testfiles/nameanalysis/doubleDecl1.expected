Error at line 3: symbol 'a' is already declared!
Error at line 5: symbol 'a' is already declared!
Error at line 6: symbol 'b' is already declared!
Error at line 8: symbol 'main' is already declared!
